packages/config-plugins/src/utils/__tests__/XML-test.ts

```diff
@@ -1,52 +1,90 @@
+import fs from 'fs-extra';
 import { vol } from 'memfs';
 
 import { buildResourceItem, readResourcesXMLAsync } from '../../android/Resources';
 import { setStringItem } from '../../android/Strings';
-import { escapeAndroidString, format } from '../XML';
+import { escapeAndroidString, format, unescapeAndroidString, writeXMLAsync } from '../XML';
 
 jest.mock('fs');
 
 export const sampleStringsXML = `
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <resources>
-  <string name="app_name">expo &amp;bo&lt;y&gt;&apos;</string>
+  <string name="app_name">exp\\'o &amp;bo&lt;y&gt;&apos;</string>
```

Comment:
Why reduce the complexity of the input test string?

Reply:
what do you mean reduce? adding `\'` complicates this example

If you are talking about why I didn't use this in that other test, then it's because test would not pass:
- `<?xml version="1.0" encoding="UTF-8" standalone="yes"?>` was removed when reading and writing
- there was sth wrong with `&amp;bo&lt;y&gt;&apos;` part but I don't remember exactly.

As I mentioned in a PR description this is not a proper fix, I currently don't have the bandwidth to work on that, so I'm just patching this case. 

btw we have a similar change in eas-build code directly and I know this is a hacky solution, so it is fine from our side if you don't want to merge it.

Reply:
Yes I meant to have the original test be complicated as a way to ensure a series of issues ([including this](https://github.com/expo/expo-cli/issues/3998)) would be fixed, if you change the functionality and the tests break then it potentially means the fixes are broken too.

Reply:
Ok I'm still not sure what you want me to do here, do you want me to leave `sampleStringsXML` as it was?

> series of issues (https://github.com/expo/expo-cli/issues/3998)

this PR is a fix for that exact case, if I read XML using `readXMLAsync` and write it using `writeXMLAsync` the file will be malformed if the name contained an apostrophe. (caused by this code https://github.com/expo/expo-cli/blob/main/packages/config-plugins/src/utils/XML.ts#L52)

> if you change the functionality and the tests break then it potentially means the fixes are broken too.

my changes did not break this test, I added additional `\'` just to complicate it more(if that is a problem I can remove it)

The part I mentioned that did not work is if I use this example in the new test I added that reads and writes XML(using `readXMLAsync/writeXMLAsync`), but this probably never worked.